### ISSUE TITLE
Does not find versioned libraries

### ISSUE THREAD

**Issue by benz0li** [JuliaGPU/CUDA_Runtime_Discovery.jl#10]
Discussion: https://github.com/JuliaGPU/CUDA.jl/discussions/2318

**Comment by benz0li** [comment]
Workarounds: Either

* install the `-dev` packages of the required libraries

or

* create unversioned symlinks for all required libraries

i.e.

```bash
libDir=/usr/local/cuda/lib64
for f in "$libDir/"*; do
  [ -e "$f" ] || [ -L "$f" ] || continue
  src="$(basename "$f" | grep "^lib.*.so.[[:digit:]]\+$")"
  if [ -s "$libDir/$src" ] && [ ! -e "$libDir/${src%.*}" ]; then
    ln -svT "$src" "$libDir/${src%.*}"
  fi
done
```

Ping @maleadt

**Comment by maleadt** [comment]
Should be fixed by https://github.com/JuliaGPU/CUDA_Runtime_Discovery.jl/commit/e2c157bf6fa205f79214ef8207b1df9ffadf7433; can you try? I verified locally with a runtime container + cuda-cupti + cuda-nvcc + cuda-command-line-tools.

FYI, it should be possible to make `ptxas` or even `nvdisasm` optional and rely on the driver to assemble code, in case you would require a more lightweight container (seeing how cuda-nvcc pulls in an entire C++ development toolchain).

**Comment by benz0li** [comment]
> Should be fixed by [e2c157b](https://github.com/JuliaGPU/CUDA_Runtime_Discovery.jl/commit/e2c157bf6fa205f79214ef8207b1df9ffadf7433); can you try? I verified locally with a runtime container + cuda-cupti + cuda-nvcc + cuda-command-line-tools.

Confirmed.

> FYI, it should be possible to make `ptxas` or even `nvdisasm` optional and rely on the driver to assemble code

🆗

> in case you would require a more lightweight container (seeing how cuda-nvcc pulls in an entire C++ development toolchain).

Currently not required, because the C++ development toolchain is installed in a later [image build] stage anyway.
